plugins/catalog-backend/src/ingestion/processors/bitbucket/types.ts

```diff
@@ -13,16 +13,18 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-export type Project = {
-  key: string;
-};
+export namespace Bitbucket {
```

Comment:
We avoid namespaces and rely on prefixing instead, so this would be `BitbucketRepository` etc. Then I'm thinking we could probably inline the `Project` and `Link` types into the `Repository` one, just to avoid exporting too much stuff.

Reply:
An alternative is to not expose this either, but to pass the full path to the matching repository in the `path` parameter:

https://github.com/backstage/backstage/pull/5494/files#diff-d09419f7c65943063cd642d5b87a1704b1ac91ebe913571adf9c74022d45ad21R25

And once again move more responsibility to the implementor, while keeping the interface very generic. Whats your thoughts here @Rugvip ? Could that be a better solution?

Reply:
You mean passing the URL before `readBitbucketOrg`, or something different? If it's just a couple of lines of code added to the user implementation I think it's fine, but likely it's worth passing in this repo object. You could potentially also inline the type in the parser options, although that limits the ways that the parser function can be defined a lil bit.

Reply:
What I meant was that I send the found target to the parser.
e.g. you specify `https://bitbucket.mycompany.com/projects/*/repos/*/catalog-info.yaml` in your configuration, then the matching target url will be passed to the parser; https://bitbucket.mycompany.com/projects/backstage/repos/backstage/browse/catalog.yaml

I have updated the pull request to reflect my idea; but I can for sure keep the initial idea of passing the `BitbucketRepository` object

See https://github.com/backstage/backstage/pull/5494/files#diff-bc08625c6b0a1c58807732586ff2de7e66d7bdec8630d2fc9b9f6e5375cbc770R93